analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/RestrictDeserializedTypes.cs

```diff
@@ -28,7 +28,31 @@ public sealed class RestrictDeserializedTypes : RestrictDeserializedTypesBase
 
     protected override DiagnosticDescriptor Rule => S5773;
 
-    public override bool ShouldExecute() => true;
+    public override bool ShouldExecute()
+    {
+        var walker = new Walker();
+        walker.SafeVisit(Node);
+        return walker.Result;
+    }
+
+    private sealed class Walker : SafeCSharpSyntaxWalker
+    {
+        public bool Result { get; private set; }
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (!Result)
+            {
+                base.Visit(node);
+            }
+        }
+
+        public override void VisitInvocationExpression(InvocationExpressionSyntax node) =>
+            Result = node.NameIs(nameof(IFormatter.Deserialize));
+
+        public override void VisitObjectCreationExpression(ObjectCreationExpressionSyntax node) =>
+            Result = node.Type.NameIs("LosFormatter");
```

Comment:
Why only check for the LosFormatter and not the others?

Reply:
I checked the issue locations. For `LosFormatter`, it is object creation. For the others, it is the invocation of `Deserialize`.